### ISSUE TITLE
logits squeezing causes an error during the inference time (If the last epoch contains only one sample)

### ISSUE THREAD

**Issue by fadiabdulf** [huggingface/transformers#28095]
https://github.com/huggingface/transformers/blob/238d2e3c44366aba9dc5c770c95475765a6725cb/src/transformers/trainer.py#L3452

**Comment by fadiabdulf** [comment]
Can anyone solve this in the next release?

**Comment by amyeroberts** [comment]
Hi @fadiabdulf, 

Please follow the [issue template](https://github.com/huggingface/transformers/blob/main/.github/ISSUE_TEMPLATE/bug-report.yml) and provide information about the running environment (run `transformers-cli env` in the terminal and copy-paste the output); the error encountered including the stack trace and a minimal code snippet that can reproduce the issue.
